contracts/bn128.scrypt

```diff
@@ -0,0 +1,581 @@
+// TODO: Split FQ, FQ2 and FQ12 into seperate libs or use generics or something like that to shorted duplicated code?
+
+struct PointFQ {
+    int x;
+    int y;
+}
+
+struct PointFQ2 {
+    int[2] x;
+    int[2] y;
+}
+
+struct PointFQ12 {
+    int[12] x;
```

Comment:
Alias
`type FQ2 = int[2];

type FQ12 = int[12];`


Also use constants, not hardcoded numbers, whenever possible.

```
static const int N = 12;
static const int N1 = 13;
```